src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -200,51 +200,19 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
                 {
                     fetchMetadataFailure = ex;
 
-                    // In this case configuration was never obtained.
-                    if (_currentConfiguration == null)
-                    {
-                        if (_bootstrapRefreshInterval < RefreshInterval)
-                        {
-                            // Adopt exponential backoff for bootstrap refresh interval with a decorrelated jitter if it is not longer than the refresh interval.
-                            TimeSpan _bootstrapRefreshIntervalWithJitter = TimeSpan.FromSeconds(new Random().Next((int)_bootstrapRefreshInterval.TotalSeconds));
-                            _bootstrapRefreshInterval += _bootstrapRefreshInterval;
-                            _syncAfter = DateTimeUtil.Add(DateTime.UtcNow, _bootstrapRefreshIntervalWithJitter);
-                        }
-                        else
-                        {
-                            _syncAfter = DateTimeUtil.Add(
-                                DateTime.UtcNow,
-                                AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
-                        }
-
-                        throw LogHelper.LogExceptionMessage(
-                            new InvalidOperationException(
-                                LogHelper.FormatInvariant(
-                                    LogMessages.IDX20803,
-                                    LogHelper.MarkAsNonPII(MetadataAddress ?? "null"),
-                                    LogHelper.MarkAsNonPII(_syncAfter),
-                                    LogHelper.MarkAsNonPII(ex)),
-                                ex));
-                    }
-                    else
-                    {
-                        _syncAfter = DateTimeUtil.Add(
-                            DateTime.UtcNow,
-                            AutomaticRefreshInterval < RefreshInterval ? AutomaticRefreshInterval : RefreshInterval);
-
-                        LogHelper.LogExceptionMessage(
-                            new InvalidOperationException(
-                                LogHelper.FormatInvariant(
-                                    LogMessages.IDX20806,
-                                    LogHelper.MarkAsNonPII(MetadataAddress ?? "null"),
-                                    LogHelper.MarkAsNonPII(ex)),
-                                ex));
-                    }
+                    LogHelper.LogExceptionMessage(
+                        new InvalidOperationException(
+                             LogHelper.FormatInvariant(
+                                LogMessages.IDX20806,
```

Comment:
We used to throw IDX20803 and IDX20806, wonder if this is behavioral breaking change?

Maybe we should throw IDX20803? It's more general.

Reply:
@pmaytak  i know we logged 20806, but i don't see where we threw it.

Reply:
You are right.

nit: Do we need to log IDX20806 here? IDX20803 is thrown at the end of this method and logs the exact same exception (`fetchMetadataFailure`).

Reply:
@pmaytak We log because in case some edge case _currentConfiguration was set before we return..